packages/components/src/components/date-picker/sub-component/DayPicker.tsx

```diff
@@ -308,128 +308,163 @@ class DayPicker extends React.Component<
           aria-selected="false"
           className="tk-daypicker-day--outside"
           tabIndex={-1}
+          role="gridcell"
+          type="button"
         />
       );
     });
   }
 
-  renderBody() {
+  renderInsideDay(days: number[]): JSX.Element[] {
     const {
-      dir,
       locale,
       selectedDays,
       disabledDays,
       highlightedDays,
       onDayClick,
     } = this.props;
     const { today, currentMonth } = this.state;
+    const selectedDateString = selectedDays
+      ? lightFormat(selectedDays, 'yyyy-MM-dd')
+      : null;
+    const todayDateString = lightFormat(today, 'yyyy-MM-dd');
+
+    const isSelectedDayVisible =
+      selectedDays &&
+      differenceInCalendarMonths(selectedDays, currentMonth) === 0;
+
+    return days.map((cell) => {
+      const cellDate = setDate(currentMonth, cell);
+      const cellName = formatDay(cellDate, locale);
+
+      const itemDateString = lightFormat(cellDate, 'yyyy-MM-dd');
+      const isSelected = itemDateString === selectedDateString;
+      const isToday = itemDateString === todayDateString;
+      const isDisabled = matchDay(cellDate, disabledDays);
+      const isHighlighted = matchDay(cellDate, highlightedDays);
+      const ariaSelected = isDisabled ? undefined : isSelected;
+      const isTabIndex = isSelectedDayVisible
+        ? isSelected
+          ? 0
+          : -1
+        : isToday
+          ? 0
+          : -1; // focus on selected day otherwise current day
+      return (
+        <button
+          key={cellName}
+          className={clsx(
+            'tk-daypicker-day',
+            {
+              'tk-daypicker-day--selected': isSelected,
+            },
+            {
+              'tk-daypicker-day--today': isToday,
+            },
+            {
+              'tk-daypicker-day--highlighted': isHighlighted && !isDisabled,
+            },
+            { 'tk-daypicker-day--disabled': isDisabled }
+          )}
+          role="gridcell"
+          type="button"
+          aria-label={cellName}
+          aria-selected={ariaSelected}
+          tabIndex={isTabIndex}
+          onKeyDown={(e) =>
+            this.handleKeyDownCell(e, cellDate, {
+              disabled: isDisabled,
+              selected: isSelected,
+            })
+          }
+          onClick={() =>
+            onDayClick(cellDate, {
+              disabled: isDisabled,
+              selected: isSelected,
+            })
+          }
+        >
+          {cell}
+        </button>
+      );
+    });
+  }
+
+  renderBody() {
+    const {
+      dir,
+      locale,
+    } = this.props;
+    const { currentMonth } = this.state;
     const daysInMonth = getDaysInMonth(currentMonth);
     const daysNeededForLastMonth = getDaysNeededForLastMonth(
       currentMonth,
       locale
     );
+    const daysNeedForFirstRow = 7 - daysNeededForLastMonth;
+    const daysNeedForFirstRowArray = [...Array(daysNeedForFirstRow).keys()].map(i => i + 1);
```

Comment:
```suggestion
    const daysNeedForFirstRowArray = Array.from({ length: daysNeedForFirstRow }, (_, i) => i + 1);
```

See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from#using_arrow_functions_and_array.from